### ISSUE TITLE
Edit badge colors in settings page to have better text contrast

### ISSUE THREAD

**Issue by DNin01** [ScratchAddons/ScratchAddons#5462]
### Why this would be helpful

Currently, I can feel the reading troubles some people might have when looking at some of the badges (especially _For editor_) in the full screen settings page.
![Current badge colors](https://user-images.githubusercontent.com/106490990/208538874-97678246-0ed3-451f-9070-c927da17a870.png)

### My suggestion

My big request is to change the _For editor_ label to be brighter with black text:
![The badge colors with updated "For editor" badge](https://user-images.githubusercontent.com/106490990/208539224-03933f94-4be3-459f-bb87-2264e4c7d50a.png)
```
.badge.darkgreen {
  background: #0cd1c7;
  color: #000;
  border-color: #12a39e;
}
```
<sub>(I also sneaked in completely black text to replace the very dark gray text)

If that's not enough, we can get even higher contrast by making all the text black to make them all WCAG-compliant.
![The badge colors with all badges updated](https://user-images.githubusercontent.com/106490990/208539410-753ad23a-a376-4b70-9e02-744878985607.png)
```
.badge {
  color: #000;
}

.badge.blue {
  background: #56a9ff;
  border-color: #3c88d7;
}
.badge.green {
  background: #78fd2b;
  border-color: #53b31c;
}
.badge.yellow {
  background: #fddd2b;
  border-color: #fdab12;
}
.badge.red {
  background: #ff9031;
  border-color: #cb5e00;
}
.badge.darkred {
  background: #ff7272;
  border-color: #d53b3b;
}
.badge.darkgreen {
  background: #0cd1c7;
  border-color: #12a39e;
}
.badge.purple {
  background: #ff67de;
  border-color: #e137bd;
}
```

And by also adding this:
```
.badge:hover {
    background: #FFF;
    border-color: #DDD;
}
```
...we can make hovering over the badges basically switch them to a black text on white version, helping to make the text easy to read.

### Possible alternatives

- Make the labels darker with white text.
- Add a high contrast mode and make these optional.

### Additional context

Checked with https://contrastchecker.com/

**Comment by CST1229** [comment]
We could also make the colors darker with white text instead of making the text white:
![](https://user-images.githubusercontent.com/68464103/208591938-5eae1d6d-dec7-4f42-b64e-84f926f92abc.png)![](https://user-images.githubusercontent.com/68464103/208592134-619a658b-aa56-469c-b2a6-2324525119dd.png)![image](https://user-images.githubusercontent.com/68464103/208592275-d3b745f9-48c7-4ad7-b48a-77f792472378.png)![image](https://user-images.githubusercontent.com/68464103/208592327-a40f6a90-847f-4813-b319-a154db562676.png)![](https://user-images.githubusercontent.com/68464103/208592560-aedce851-fdbd-42c9-84ef-cc7703dd9391.png)![](https://user-images.githubusercontent.com/68464103/208592728-a7f1f1e7-e6cb-4f70-b794-f4e040c5579c.png)![image](https://user-images.githubusercontent.com/68464103/208592952-74dab637-6afa-4eb7-91b7-dbd2a3545e6e.png)

**Comment by Secret-chest** [comment]
Actual experience matters more than numbers. Same for Scratch's new blocks.

**Comment by mxmou** [comment]
Most of the tags don't really need unique colors. See #4989

**Comment by mxmou** [comment]
> Actual experience matters more than numbers. Same for Scratch's new blocks.

Better contrast makes text easier to read, which improves actual experience.

**Comment by DNin01** [comment]
> We could also make the colors darker on white-text tags instead of making the text on black-text tags white (though this makes For website and Forums look out of place with the other almost-black tags and probably looks horrible):
> 
> ![](https://user-images.githubusercontent.com/68464103/208591938-5eae1d6d-dec7-4f42-b64e-84f926f92abc.png)![](https://user-images.githubusercontent.com/68464103/208592134-619a658b-aa56-469c-b2a6-2324525119dd.png)![image](https://user-images.githubusercontent.com/68464103/208592275-d3b745f9-48c7-4ad7-b48a-77f792472378.png)![image](https://user-images.githubusercontent.com/68464103/208592327-a40f6a90-847f-4813-b319-a154db562676.png)![](https://user-images.githubusercontent.com/68464103/208592560-aedce851-fdbd-42c9-84ef-cc7703dd9391.png)![](https://user-images.githubusercontent.com/68464103/208592728-a7f1f1e7-e6cb-4f70-b794-f4e040c5579c.png)![image](https://user-images.githubusercontent.com/68464103/208592952-74dab637-6afa-4eb7-91b7-dbd2a3545e6e.png)
> 
> 

I'd take either. Maybe we could change the palette depending on the theme (whether the user is in light mode or dark mode).

**Comment by penguinmoose** [comment]
> Most of the tags don't really need unique colors. See #4989

However, if tags have unique colors, it would be easier to differentiate between them.

**Comment by penguinmoose** [comment]
I think all the text should be black, and the background color should be lightened significantly.

**Comment by BroJac5246** [comment]
> I think all the text should be black, and the background color should be lightened significantly.

I think the text color should be whatever looks best on the background.

**Comment by penguinmoose** [comment]
@BroJac5246 That's true.

**Comment by DNin01** [comment]
Depending on the set of guidelines we use, the proposed colors may not be optimal. I think they pass WCAG, but not the more accurate [APCA](https://www.myndex.com/APCA/).

**Comment by WorldLanguages** [comment]
Related discussion about WCAG vs APCA: https://twitter.com/jacobmparis/status/1672267309275127812
